src/macro.jl

```diff
@@ -65,12 +68,18 @@ ERROR: AssertionError: Float64($(Expr(:escape, :sqrt5))) == Float64(big($(Expr(:
 ```
 """
 macro irrational(sym::Symbol, val::Float64, def::Union{Symbol,Expr}, T::Symbol=Symbol(uppercasefirst(string(sym))))
-    irrational(sym, val, def, T)
+    irrational(__module__, sym, val, def, T)
 end
 macro irrational(sym::Symbol, def::Union{Symbol,Expr}, T::Symbol=Symbol(uppercasefirst(string(sym))))
-    irrational(sym, :(big($(esc(sym)))), def, T)
+    irrational(__module__, sym, :(big($(esc(sym)))), def, T)
 end
-function irrational(sym::Symbol, val::Union{Float64,Expr}, def::Union{Symbol,Expr}, T::Symbol)
+function irrational(mod::Module, sym::Symbol, val::Union{Float64,Expr}, def::Union{Symbol,Expr}, T::Symbol)
+    if isdefined(mod, T)
+        throw(ArgumentError("Type `$T` of irrational constant `$sym` is already defined in module `$mod`."))
```

Comment:
Now that we only support Julia >= 1.10 anymore, we can use lazy strings for error messages (which only build the string and interpolate the string representation of `T` etc. when the error message is shown):
```suggestion
        throw(ArgumentError(LazyString("Type `", T, "` of irrational constant `", sym, "` is already defined in module `", mod, "`.")))
```